File: Factorial.cs
public class Factorial {
	public static void Main(string[] args) {
		System.Func<long, long> factorial = null;
		factorial = n => (n==0) ? 1 : n*factorial(n-1);
		System.Console.WriteLine(factorial(System.Convert.ToInt64(args[0])));
	}
}

File: Factorial.java
public class Factorial<I> {
	public I func;

	public static void main(String[] args) {
		Factorial<Operation> factorial = new Factorial<>();
		factorial.func = (n) -> ((n==0) ? 1 : n*factorial.func.op(n-1));
		System.out.println(factorial.func.op(Long.parseLong(args[0])));
	}

	public interface Operation {
		long op(long a);
	}
}